mlflow/utils/databricks_utils.py

```diff
@@ -954,6 +953,10 @@ def _get_databricks_creds_config(tracking_uri):
                 break
 
     if not config or not config.host:
+        if ignore_error:
+            # If we use Databricks sdk to do authentication,
+            # we can ignore error in `_get_databricks_creds_config`
+            return config or DatabricksConfig.empty()
         _fail_malformed_databricks_auth(tracking_uri)
 
     return config
```

Comment:
what would happen if we remove `_fail_malformed_databricks_auth` and always fallback to databricks SDK?

Reply:
we can't. `get_databricks_env_vars` is still used in many places (majorly for setting subprocess environment for authentication). `get_databricks_env_vars` gets credential values by calling `_get_databricks_creds_config`. So we should raise error if it can't get valid credential values